packages/hash/frontend/src/pages/[shortname]/types/entity-type/[entity-type-id].page/definition-tab/property-list-card/shared/property-type-form/expected-value-selector/custom-expected-value-builder/array-expected-value-builder/array-min-max-items.tsx

```diff
@@ -0,0 +1,302 @@
+} from "react";
+import { useController, useFormContext } from "react-hook-form";
+
+import { PropertyTypeFormValues } from "../../../../property-type-form-values";
+
+type ItemInputProps = { width: string } & InputProps;
+
+const ItemInput = forwardRef(({ width, ...props }: ItemInputProps, ref) => (
+  <Input
+    type="number"
+    {...props}
+    ref={ref}
+    sx={({ palette, transitions }) => ({
+      width,
+      fontSize: 11,
+      height: 16,
+      color: palette.white,
+      transition: transitions.create("color"),
+
+      "::before": {
+        display: "none",
+      },
+      "::after": {
+        borderBottomStyle: "dotted",
+        borderBottomColor: palette.white,
+        borderBottomWidth: 1,
+      },
+
+      ":hover": {
+        "::after": {
+          transform: "scaleX(1) translateX(0)",
+        },
+      },
+
+      [`.${inputClasses.disabled}`]: {
+        WebkitTextFillColor: "initial",
+      },
+
+      // <-- Hide number input default arrows -->
+      "& input[type=number]": {
+        WebkitAppearance: "textfield",
+      },
+      "& input[type=number]::-webkit-outer-spin-button": {
+        WebkitAppearance: "none",
+        margin: 0,
+      },
+      "& input[type=number]::-webkit-inner-spin-button": {
+        WebkitAppearance: "none",
+        margin: 0,
+      },
+      // <-- Hide number input default arrows -->
+    })}
+  />
+));
+
+interface ArrayMinMaxItemsProps {
+  arrayId: string;
+}
+
+export const ArrayMinMaxItems: FunctionComponent<ArrayMinMaxItemsProps> = ({
+  arrayId,
+}) => {
+  const { control, setValue } = useFormContext<PropertyTypeFormValues>();
+
+  const minItemsController = useController({
+    control,
+    name: `flattenedCustomExpectedValueList.${arrayId}.data.minItems`,
+  });
+  const maxItemsController = useController({
+    control,
+    name: `flattenedCustomExpectedValueList.${arrayId}.data.maxItems`,
+  });
+  const infinityController = useController({
+    control,
+    name: `flattenedCustomExpectedValueList.${arrayId}.data.infinity`,
+  });
+
+  const [minItemsWidth, setMinItemsWidth] = useState(0);
+  const [maxItemsWidth, setMaxItemsWidth] = useState(0);
+  const [hovered, setHovered] = useState(false);
+
+  const minItemsInputRef = useRef<HTMLInputElement | null>();
+  const maxItemsInputRef = useRef<HTMLInputElement | null>();
+
+  const setInputSize = useCallback(() => {
+    setMinItemsWidth(minItemsInputRef.current?.value.length ?? 1);
+    setMaxItemsWidth(maxItemsInputRef.current?.value.length ?? 1);
+  }, []);
+
+  useEffect(() => {
+    setInputSize();
+  }, [setInputSize]);
+
+  return (
+    <Box display="flex" gap={1.25}>
+      <Box display="flex">
+        <Box
+          sx={{
+            display: "inline-flex",
+            alignItems: "center",
+            py: 0.25,
+            px: 1,
+            background: ({ palette }) => palette.gray[80],
+            borderBottomLeftRadius: 30,
+            borderTopLeftRadius: 30,
+          }}
+        >
```

Comment:
Instead of defining all 4 different border radii on two different elements, you can give their wrapper `borderRadius: x, overflow: hidden`

Reply:
That's cleaner thanks (f968f7f89fd387b8b526275af0e845359d621730)

Reply:
@luisbettencourt Did you checked if `borderRadius: 30` works as expected? If I remember correct, MUI interprets `borderRadius: number` different than `borderBottomLeftRadius: number`. 

Like how it interprets `ml: 3` as `ml: 24px` (3 is multiplier if you don't explicitly say `3px`)

Reply:
You're right. Fixed it here: 6691ac15f64c9c9345fafcaadaafb6816d0ddac1